generators/profilegen/src/profile_generator.rb

```diff
@@ -0,0 +1,225 @@
+        end
+
+        generate_model_types(resource_provider, resource_type_version)
+      end
+
+      file = get_module_file resource_provider
+      file.write(get_renderer(ProfileTemplates.module_template))
+      @model_types, @operation_types, @versions_clients_mapper = [], [], {}
+      @management_client = ''
+    end
+  end
+
+  #
+  # Method to generate the client file
+  # Gets the client template and renders it
+  #
+  def generate_client
+    file = get_client_file
+    file.write(get_renderer(ProfileTemplates.client_template))
+  end
+
+  def generate_module_definition
+    file = get_module_definition_file
+    file.write(get_renderer(ProfileTemplates.module_definition_template))
+  end
+
+  #
+  # Method to get renderer
+  #
+  # @param template for which the renderer will be created
+  #
+  def get_renderer(template)
+    renderer = ERB.new(template, 0, '-%>')
+    renderer.result(get_binding)
+  end
+
+  def generate_operation_types(resource_provider, resource_type, resource_type_version)
+    namespace = @dir_metadata[resource_provider]['namespace'] + "::#{get_version(resource_type_version)}"
+    namespace = namespace.gsub(/\./, '_' )
+    obj = Module.const_get(namespace)
+    operations = []
+    obj.constants.select do |const_name|
+      if((obj.const_get(const_name).instance_of?Class))
+        if(const_name.to_s.end_with?('ManagementClient') || const_name.to_s == 'ManagementLinkClient' || const_name.to_s == 'ManagementLockClient' || const_name.to_s == 'FeatureClient' || const_name.to_s == 'PolicyClient' || const_name.to_s == 'SubscriptionClient')
+          @management_client = obj.name + '::' + const_name.to_s
+          @versions_clients_mapper[resource_type_version] = @management_client
+        elsif (const_name.to_s == resource_type || '*' == resource_type)
+          operations << const_name.to_s
+        end
+      end
+    end
+
+    if operations.empty?
+      raise "#{resource_type} operation could not be found for RP: #{resource_provider}:Version: #{resource_type_version}"
+    end
+
+    operations.each do |operation|
+      check_and_add_operation(obj, operation, resource_provider)
+    end
+  end
+
+  def check_and_add_operation(module_obj, operation, resource_provider)
+    operation_body = module_obj.name + '::' + operation
+    operation_name_ruby = get_model_name(operation)
+
+    if(check_for_availability(@operation_types, operation_name_ruby))
+      raise "#{operation_name} operation is appearing twice for the RP: #{resource_provider}."
+    else
+      @operation_types << { 'operation_name': operation, 'operation_name_ruby': operation_name_ruby, 'operation_body': operation_body}
+    end
+  end
+
+  def generate_model_types(resource_provider, resource_type_version)
+    namespace = @dir_metadata[resource_provider]['namespace'] + + "::#{get_version(resource_type_version)}::Models"
+    namespace = namespace.gsub(/\./, '_' )
```

Comment:
please document why are we replacing `.` with `_` and what is the assumption here

Reply:
Done